src/app/components/Curation/CurationGrid/index.tsx

```diff
@@ -33,35 +52,58 @@ const CurationGrid = ({
     ...eventTrackingData,
   });
 
+  const renderPromo = (promo: Summary, index: number) => {
+    const isFirstPromo = index === 0;
+    const lazyLoadImages = !(isFirstPromo && isFirstCuration);
+    const service = extractWorldServiceFromUrl(promo.link);
+
+    if (isHighImpact(promo) && !isMediaType(promo) && !isLite) {
```

Comment:
Can we somehow reuse the variable we have already calculated, rather than duplicating? Perhaps pass it into the renderPromo as a prop?

```suggestion
    if (hasHighImpactPromo && !isLite) {
```

Reply:
Or could we do something like this

```
const Promo = isHighImpact ? HighImpactPromo : CurationPromo;

return (
          <ServiceContextProvider service={service}>
            <Promo
              {...promo}
              lazy={lazyLoadImages}
              eventTrackingData={buildPromoEventTrackingData(promo, index)}
            />
          </ServiceContextProvider>
        );
```

And there would be no harm in always rendering the ServiceContextProvider, even if the service is null? (Or could the fallback be the original service from the service context?)

Reply:
We have this duplication because potentially there might be a mix of High Impact and regular promos. If we used `hasHighImpactPromo` within the `renderPromo`, then standard promo would be rendered and high-impact. 

Probably we could always use ServiceContextProvider, but I'm just wondering about some unexpected behavior when using existing standard curations. 🤔 

Eg.. if by some chance the URL contained different service (i.e. `promo.link`) then within the `CurationPromo` wrong translations would be shown. I would render ServiceContextProvider only for `HighImpactPromo`.

I refactor it to make it more readable. Let me know if that works